renpy.el

```diff
@@ -1981,7 +1987,85 @@ Use enclosing keywords to find image name symbol bounds.  Return a (BEG
 			 ;; TODO: Once we know the precise context, other
 			 ;; completion engines should not apply.
 			 :exclusive 'no)))))
-
+
+;;;; Flymake
+
+(defvar renpy--flymake-regex-list
+  (list
+   ;; Example: path/to/scrip.rpy:111 Error message
+   (rx bol
+       (group (1+ (not (any ?: "\n"))))
+       ":"
+       (group (1+ digit))
+       " "
+       (group (0+ (not "\n")))
+       eol)
+   ;; Example: File \"path/to/file/script.rpy\", line 27: expected statement.
+   (rx "File "
+       "\""
+       (group (1+ (not "\"")) )
+       "\", line "
+       (group (1+ digit))
+       ": "
+       (group (0+ (not "\n")))
+       eol))
+  "Regexps used to catch renpy lint errors/warnings.")
+
+(defvar-local renpy--flymake-proc nil
+  "Flymake process for the current buffer.")
+
+(defun renpy--find-project-root ()
+  "Find the Ren'Py project root by looking for a \"game/\" directory.
+Return the full path to the \"game/\" directory if found, or nil if not."
+  (when-let (dir (locate-dominating-file default-directory "game"))
+    (expand-file-name "game" dir)))
+
+(defun renpy--parse-lint-buffer (buffer)
+  "Parse warning/error messages from BUFFER output for Flymake.
+Return a list of Flymake diagnostics."
+  (with-current-buffer buffer
+    (goto-char (point-min))
+    (let (diags)
+      (dolist (re renpy--flymake-regex-list)
+	(while (re-search-forward re nil t)
+          (let* ((fname (match-string 1))
```

Comment:
I would use `filename` instead of `fname` to avoid confusing it for "function name".
```diff
--- a/renpy.el
+++ b/renpy.el
@@ -2028,11 +2028,11 @@ defun renpy--parse-lint-buffer
     (let (diags)
       (dolist (re renpy--flymake-regex-list)
 	(while (re-search-forward re nil t)
-          (let ((fname (match-string 1))
+          (let ((filename (match-string 1))
 		(line (string-to-number (match-string 2)))
 		(msg (match-string 3)))
             (push (flymake-make-diagnostic
-                   fname (cons line 0) nil :error msg)
+                   filename (cons line 0) nil :error msg)
                   diags))))
       diags)))
 
```